### ISSUE TITLE
Email verification link for admin user does not work.

### ISSUE THREAD

**Issue by prinzdezibel** [reactioncommerce/reaction#3714]
### Expected behavior
When defining the environment variable REACTION_SECURE_DEFAULT_ADMIN=true and start a stock Reaction with empty database, a new admin user should be created. Depending on wether a mail  host is defined or not the console should show a verification link (or sent it to the mailhub respectively), which can be copied into the browser's address bar. After loading that link, the user's email should be verified (in database)

### Actual behavior
When pasting the verification link into the browser's address bar, the index route loads. There's no visual feedback or anything that signals the user if the verification process was successful or not.
In fact, the verfication does not happen at all, because the email address is still unverified in database.

### Steps to reproduce the behavior
1) Clone reaction
2) export REACTION_SECURE_DEFAULT_ADMIN=true 
3) reaction reset -n && reaction
4) Watch out in console window for something along the lines
```
  ***************************************************
            IMPORTANT! EMAIL VERIFICATION LINK

             Email sending is not configured.

    Go to the following URL to verify email: hpwgdbe3@localhost

    http://tsp.dyn.projekt-und-partner.de:3000/#/verify-email/OLzrJJbwW_Bb7KcIz1hufcdH9GxdDX9C7j7zwl80Smw
    ***************************************************
```

5) copy the link adress and paste it in the browser window
6) See nothing happen – apart from the index page being loaded
6) Browse mongodb. Both Accounts and Users collection will still show the unverified email addresses. Correct would be: `Accounts.emails.$.verified : true` and `Users.emails.$.verified: true`

**Comment by prinzdezibel** [comment]
I realized that this is because the above link seems wrong. I don't know if this is because it's a hash-based URL, but it seems like the URL is a no-op (maybe it has to do with the react router not supporting hash-based URLs, I don't know)

The URL scheme that actually is working would be:

http://tsp.dyn.projekt-und-partner.de:3000/account/profile/verify?email=testing%40localhost

**Comment by prinzdezibel** [comment]
Talked with Brent about it. The email verification is sort of an esoteric feature which presumably isn't used a whole lot. The provided PR does fix it in a way we are doing it for the non-admin accounts as well. Therefore it doesn't add new security concerns. We agreed that the current state with email encoded links isn't a good thing, but it might be worth to create a new ticket for that and treat this PR separately from any security related task.

**Comment by zenweasel** [comment]
For background here is @jshimko's comment:
> So I agree that this was an issue, but this strategy raises some security concerns for how this was being implemented for non-admin accounts in general. Once someone sees that the verification link simply requires a URL encoded email (instead of a unique token), what would stop someone from verifying any email they want by manually changing the URL to a known account's email address?

> The original Meteor accounts implementation avoided that potential security concern by creating a random token that gets used in the URL and it is then used to securely identify the user upon visiting the link. So perhaps that got broken at some point and/or UI notifications were never added to make it clear when it worked, but I think that is clearly the correct/secure way to verify an email. Having an array of tokens also allows you to have expirations on specific verification URL's (a fairly common practice).

Not sure where this fits in priority-wise so I think it makes sense to create a ticket to address the potential security issue for all of these and just fix this particular fix in for now.

**Comment by zenweasel** [comment]
Closed via #3884
